src/corva/models/stream/raw.py

```diff
@@ -145,7 +148,7 @@ def filter_records(
     ) -> List[RawBaseRecord]:
         new_records = copy.deepcopy(self.records)
 
-        if self.is_completed:
+        if self.is_completed and new_records:
```

Comment:
Maybe we don't need this check as logic below will succeed for empty lists too?
```suggestion
        if self.is_completed:
```

Reply:
You are right, we do not need it. Code [is updated](https://github.com/corva-ai/python-sdk/pull/86/commits/136c4f37db617749eda8fe8d5a4a2f1bb0820308).